src/playbooks/with_build/vsd_standby_upgrade_deploy.yml

```diff
@@ -0,0 +1,145 @@
+---
+- hosts: primary_vsds
+  gather_facts: no
+  remote_user: "{{ vsd_default_username }}"
+  tasks:
+    - name: Run vsd-prepare-replication-master-cluster
+      command: "/opt/vsd/bin/vsd-prepare-replication-master-cluster.sh"
+      when: vsd_standby | default(false)
+
+- hosts: vsd_ha_node1
+  gather_facts: no
+  remote_user: "{{ vsd_default_username }}"
+  tasks:
+    - name: Clear the backup dir of old backups
+      file:
+        path: "/tmp/backup"
+        state: absent
+      when:
```

Comment:
This same "when" clause is used for all tasks.  Can you use a block to combine them all?  Also, this is quite a few tasks for a playbook.  Can you consider moving these to a role?

Reply:
Added a block. Tested and ready for review.

Reply:
Moving to a role would be tricky as the tasks alternate between hosts

Reply:
No, just the tasks for this host.  It is bigger than the rest.